### DISCUSSION TITLE
How do I extract files?

### DISCUSSION THREAD

**Discussion by Chrisgraue** [SilverGreen93/CDPExplorer#7] [General]
I've downloaded this to try and extract some sound files from some CDPs I got for Trainz. I've opened the software, added the CDP file, but I'm not seeing it open up the list of files that are within the CDP. 

According to the file list for the CDP, it appears there should be multiple wav files in it.

This is one of the files I'm trying to open: https://kuid.trainz-mp.ru/index.php?kuid=0005b79c0001874800

I've attached a screenshot of what I'm seeing in the app 
![cdpScreenshot](https://github.com/SilverGreen93/CDPExplorer/assets/117547373/862a7ea5-d627-4b6d-99b9-c5b2ea8067c3)

**Comment by SilverGreen93** [comment]
Hi. CDP Explorer allows you to extract individual assets from a CDP file containing multiple assets.
In your case, the CDP file contains a single asset.
To extract the individual files from within an asset you must install it in Trainz itself. This software does not permit extracting of files from the assets from within the CDP file.

**Reply by RebbyGamerBoi** [reply]
Can You Add with Extract Files with folders?

**Reply by SilverGreen93** [reply]
> Can You Add with Extract Files with folders?

What do you mean by Extact with folders?

**Reply by RebbyGamerBoi** [reply]
like when you extract iit will allow you to do as folders

**Reply by RebbyGamerBoi** [reply]
with the assets you need in there not cdp

**Reply by SilverGreen93** [reply]
There is no possibility to use CDP explorer to extract the individual files from CDP file. You must use any version of Trainz to install the asset and extract the files from there.
